tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java

```diff
@@ -143,22 +396,134 @@ public void testListGroupCommand(String quorum) throws Exception {
             return out.get().contains("STATE") && out.get().contains(simpleGroup) && out.get().contains(GROUP);
         }, "Expected to find " + simpleGroup + ", " + GROUP + " and the header, but found " + out.get());
 
-        String[] cgcArgs3 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "Stable"};
+        String[] cgcArgs3 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--type"};
         TestUtils.waitForCondition(() -> {
             out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
                 ConsumerGroupCommand.main(cgcArgs3);
                 return null;
             }));
+            return out.get().contains("TYPE") && !out.get().contains("STATE") && out.get().contains(simpleGroup) && out.get().contains(GROUP);
+        }, "Expected to find " + simpleGroup + ", " + GROUP + " and the header, but found " + out.get());
+
+        String[] cgcArgs4 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "--type"};
+        TestUtils.waitForCondition(() -> {
+            out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs4);
+                return null;
+            }));
+            return out.get().contains("TYPE") && out.get().contains("STATE") && out.get().contains(simpleGroup) && out.get().contains(GROUP);
+        }, "Expected to find " + simpleGroup + ", " + GROUP + " and the header, but found " + out.get());
+
+        String[] cgcArgs5 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "Stable"};
+        TestUtils.waitForCondition(() -> {
+            out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs5);
+                return null;
+            }));
             return out.get().contains("STATE") && out.get().contains(GROUP) && out.get().contains("Stable");
         }, "Expected to find " + GROUP + " in state Stable and the header, but found " + out.get());
 
-        String[] cgcArgs4 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "stable"};
+        String[] cgcArgs6 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "stable"};
         TestUtils.waitForCondition(() -> {
             out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
-                ConsumerGroupCommand.main(cgcArgs4);
+                ConsumerGroupCommand.main(cgcArgs6);
                 return null;
             }));
             return out.get().contains("STATE") && out.get().contains(GROUP) && out.get().contains("Stable");
         }, "Expected to find " + GROUP + " in state Stable and the header, but found " + out.get());
+
+        String[] cgcArgs7 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--type", "Classic"};
```

Comment:
I was finally able to merge https://github.com/apache/kafka/pull/15382 before this one. Could you please adapt those tests?

Reply:
of course